src/Tools/dotnet-monitor/Egress/EgressService.cs

```diff
@@ -16,31 +21,129 @@ namespace Microsoft.Diagnostics.Tools.Monitor.Egress
     /// <summary>
     /// Egress service implementation required by the REST server.
     /// </summary>
-    internal class EgressService : IEgressService
+    internal class EgressService : IEgressService, IDisposable
     {
-        private readonly IOptionsMonitor<EgressOptions> _egressOptions;
+        private readonly IEnumerable<IEgressProviderConfigurationProvider> _providers;
+        private readonly IDictionary<string, string> _providerTypeMap;
+        private readonly IDictionary<string, Type> _optionsTypeMap;
+        private readonly IList<IDisposable> _registrations;
+        private readonly IServiceProvider _serviceProvider;
 
-        public EgressService(IOptionsMonitor<EgressOptions> egressOptions)
+        public EgressService(
+            IServiceProvider serviceProvider,
+            IEnumerable<IEgressProviderConfigurationProvider> providers)
```

Comment:
Also get IEgressPropertiesConfigurationProvider and listen for its change token.

Reply:
Fixed by listening to change token from the Egress configuration section rather than individual egress configuration subsections.